snapcraft/plugins/maven.py

```diff
@@ -103,18 +114,22 @@ def build(self):
 
         self.run(mvn_cmd + self.options.maven_options)
 
-        jarfiles = glob.glob(os.path.join(self.builddir, 'target', '*.jar'))
-        warfiles = glob.glob(os.path.join(self.builddir, 'target', '*.war'))
-        if not (jarfiles or warfiles):
-            raise RuntimeError('could not find any built jar files for part')
-        if jarfiles:
-            jardir = os.path.join(self.installdir, 'jar')
-            os.makedirs(jardir, exist_ok=True)
-            self.run(['cp', '-a'] + jarfiles + [jardir])
-        if warfiles:
-            wardir = os.path.join(self.installdir, 'war')
-            os.makedirs(wardir, exist_ok=True)
-            self.run(['cp', '-a'] + warfiles + [wardir])
+        for f in self.options.maven_targets:
+            src = os.path.join(self.builddir, f, 'target')
+
+            jarfiles = glob.glob(os.path.join(src, '*.jar'))
+            warfiles = glob.glob(os.path.join(src, '*.war'))
+            arfiles = glob.glob(os.path.join(src, '*.[jw]ar'))
+
+            if not (arfiles):
+                raise RuntimeError("could not find any"
+                                   "built jar files for part")
+            basedir = 'jar' if jarfiles and len(f) == 0 else (
+                      'war' if warfiles and len(f) == 0 else f)
+
+            targetdir = os.path.join(self.installdir, basedir)
+            os.makedirs(targetdir, exist_ok=True)
+            self.run(['cp', '-a'] + arfiles + [targetdir])
```

Comment:
El 07/07/16 a las 16:35, Kyle Fazzari escribió:
> In snapcraft/plugins/maven.py
> <https://github.com/snapcore/snapcraft/pull/627#discussion_r69972186>:
> 
>> +        for f in self.options.maven_targets:
>> +            src = os.path.join(self.builddir, f, 'target')
>> +
>> +            jarfiles = glob.glob(os.path.join(src, '*.jar'))
>> +            warfiles = glob.glob(os.path.join(src, '*.war'))
>> +            arfiles = glob.glob(os.path.join(src, '*.[jw]ar'))
>> +
>> +            if not (arfiles):
>> +                raise RuntimeError("could not find any"
>> +                                   "built jar files for part")
>> +            basedir = 'jar' if jarfiles and len(f) == 0 else (
>> +                      'war' if warfiles and len(f) == 0 else f)
>> +
>> +            targetdir = os.path.join(self.installdir, basedir)
>> +            os.makedirs(targetdir, exist_ok=True)
>> +            self.run(['cp', '-a'] + arfiles + [targetdir])
> 
> I'd prefer to use the python utilities for copying files rather than
> shelling out.

Ah, nice catch!

> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/snapcore/snapcraft/pull/627/files/c8dc9a6288c6c68c11da4f5c45e3e4b49493bae0#r69972186>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABF_5QIHLRzqaA1js3VMss5evoyqw9icks5qTVUbgaJpZM4JD7dL>.
>